x-pack/plugins/observability_solution/apm/server/routes/alerts/rule_types/error_count/get_error_count_chart_preview.ts

```diff
@@ -111,6 +111,12 @@ export async function getTransactionErrorCountChartPreview({
     return acc;
   }, {} as BarSeriesDataMap);
 
+  resp.aggregations.timeseries.buckets.forEach((bucket) => {
+    if (bucket.series.buckets.length === 0) {
+      Object.keys(seriesDataMap).forEach((key) => seriesDataMap[key].push({ x: bucket.key, y: 0 }));
+    }
+  });
```

Comment:
This looks odd. Would you be able to do this in the reduce function above?

Reply:
You are right, this was not the best solution. Due to how the nested aggs were used in the query, it was not possible to include empty buckets in `reduce` function. I have modified the order of nested aggs in 327b9b1d43b673240dd16d5489592cacc223e524 and removed the code where I was adding empty buckets manually after `reduce` function.